Support the use of a `length` field for `Range`

skeleten opened the pull request:
The `textDocument/didChange` has an optional `rangeLength` field. This
was not previously represented in the `Range` struct, but might be used
in `rls_vfs` to handle this event better in case the `end` field is
scrambled (looking at you, emacs)

related to: rust-lang-nursery/rls#281 and nrc/rls-vfs#12

Commit:
Support the use of a `length` field.

The `textDocument/didChange` has an optional `rangeLength` field. This
was not previously represented in the `Range` struct, but might be used
in `rls_vfs` to handle this event better in case the `end` field is
scrambled (looking at you, emacs)

nrc commented on src/lib.rs:
```diff
@@ -31,7 +31,6 @@ pub mod compiler;
 
 #[derive(Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
 pub struct Column<I: Indexed>(pub u32, PhantomData<I>);
-
```
nit: please don't remove this blank line

nrc commented on src/lib.rs:
```diff
@@ -225,6 +225,22 @@ impl<I: Indexed> Range<I> {
+        }
+    }
+
+    pub fn new_with_len(row_start: Row<I>,
```
we should be consistent about naming - use either `len` or `length` for both the field and in the methods.

nrc reviewed (approved):
A couple of minor changes, but otherwise looks good.

sophiajt commented:
Thanks!

Commit:
Consistently use `len`.

Mainly because I already opened up a PR on rust-lang-nursery/rls that
uses a `_len` method :P
